packages/react-reconciler/src/__tests__/ReactHooksWithNoopRenderer-test.js

```diff
@@ -3252,6 +3254,681 @@ describe('ReactHooksWithNoopRenderer', () => {
     });
   });
 
+  // @gate enableUseResourceEffectHook
+  describe('useResourceEffect', () => {
+    class Resource {
+      isDeleted: false;
+      id: string;
+      opts: mixed;
+      constructor(id, opts) {
+        this.id = id;
+        this.opts = opts;
+      }
+      update(opts) {
+        if (this.isDeleted) {
+          console.error('Cannot update deleted resource');
+          return;
+        }
+        this.opts = opts;
+      }
+      destroy() {
+        this.isDeleted = true;
+      }
+    }
+
+    // @gate enableUseResourceEffectHook
+    it('validates create return value', async () => {
+      function App({id}) {
+        useResourceEffect(() => {
+          Scheduler.log(`create(${id})`);
+        }, [id]);
+        return null;
+      }
+
+      await expect(async () => {
+        await act(() => {
+          ReactNoop.render(<App id={1} />);
+        });
+      }).toErrorDev(
+        'useResourceEffect must provide a callback which returns a resource. ' +
+          'If a managed resource is not needed here, use useEffect. Received undefined',
+        {withoutStack: true},
+      );
+    });
+
+    // @gate enableUseResourceEffectHook
+    it('validates non-empty update deps', async () => {
+      function App({id}) {
+        useResourceEffect(
+          () => {
+            Scheduler.log(`create(${id})`);
+            return {};
+          },
+          [id],
+          () => {
+            Scheduler.log('update');
+          },
+          [],
+        );
+        return null;
+      }
+
+      await expect(async () => {
+        await act(() => {
+          ReactNoop.render(<App id={1} />);
+        });
+      }).toErrorDev(
+        'useResourceEffect received a dependency array with no dependencies. ' +
+          'When specified, the dependency array must have at least one dependency.',
```

Comment:
At this point they have to be specified, right? Should we update message and also throw an error if no argument is provided?

Reply:
It doesn't make sense to have an update callback without any deps since that would only run on mount/unmount, hence this check. Responded to your other comment